components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -62,6 +78,13 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
   public removeText = 'Remove User';
 
   public unassigned = ProjectConstants.UNASSIGNED_PROJECT_ID;
+  public projects: ProjectMap = {};
+  // keep a list of projects that were in the teams list that are
+  // left to fetch so we can know if we are fully loaded or not.
+  // don't want to enable the projects-dropdown until we have
+  // checked the proper project checkboxes corresponding to projects
+  // already in the team.
+  public teamProjectsLeftToFetch: string[] = [];
```

Comment:
🤔 	Wondering if you could avoid the complications of this mechanism--coupled with _n_ calls to GetProject--by instead just doing a single `ListProjects`? I believe that would also address the "project not found" error I reported in our team channel, because it just naturally gets the valid projects.

Reply:
In fact, the prior art for this is in the same file. We fire GetUsers and GetTeamUsers, then do a `combineLatest` on `users, uStatus, tUsers, tStatus` to mesh the two together. I suspect we could do something similar here.